sympy/core/tests/test_args.py

```diff
@@ -591,6 +591,23 @@ def test_sympy__calculus__util__AccumulationBounds():
     assert _test_args(AccumulationBounds(0, 1))
 
 
+def test_sympy__sets__ordinals__OmegaPower():
+    from sympy.sets.ordinals import OmegaPower
+    from sympy.core.numbers import Integer
+    assert _test_args(OmegaPower(1, Integer(1)))
```

Comment:
The purpose of `test_args` module is to make sure that the arguments *created by the class to test* are instances of `Basic`. So the proper place to convert the argument is `OmegaPower.__new__`. I would write that as follows:

    if isinstance(b, int):   # Other integer types could possibly be admitted
        b = Integer(b)
    if not isinstance(b, Integer) or b < 0:
        raise TypeError(...)